src/collective/mustread/db.py

```diff
@@ -29,12 +29,26 @@ def getEngine(conn_string=None, req=None):
             audit_conn_string = registry['collective.auditlog.interfaces.IAuditLogSettings.connectionstring']  # noqa
         except KeyError:
             audit_conn_string = None
-        if conn_string == audit_conn_string \
-           and req and 'sa.engine' in req.environ:
-                # re-use collective.auditlog connector if possible
-                engine = req.environ['sa.engine']
+        if conn_parameters is None:
+            conn_parameters = registry['collective.mustread.interfaces.IMustReadSettings.connectionparameters']  # noqa
```

Comment:
You could conceivably avoid this #noqa ugliness by storing the long ids in helper vars. Not a big deal.